google/cloud/bigquery/v2/minimal/internal/job_logging.cc

```diff
@@ -25,6 +26,25 @@ namespace cloud {
 namespace bigquery_v2_minimal_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+template <typename Functor, typename Request,
+          typename Result = google::cloud::internal::invoke_result_t<
+              Functor, rest_internal::RestContext&, Request const&>>
+Result JobLogWrapper(Functor&& functor, rest_internal::RestContext& context,
+                     Request const& request, char const* where,
+                     TracingOptions const& /*options*/) {
+  auto formatter = [](std::string* out, auto const& header) {
+    absl::StrAppend(out, "name=", header.first, ", value={",
+                    absl::StrJoin(header.second, "&"), "}");
+  };
+  GCP_LOG(DEBUG) << where << "() " << request << ", Context={"
+                 << absl::StrJoin(context.headers(), ", ", formatter) << "}";
+
+  auto response = functor(context, request);
+  GCP_LOG(DEBUG) << where << "() >> status=" << response.status();
```

Comment:
We should log the response, when available ... that's half the purpose of the logging stub.